core/src/G3Timesample.cxx

```diff
@@ -148,6 +154,48 @@ G3TimesampleMap G3TimesampleMap::Concatenate(const G3TimesampleMap other)
 }
 
 
+// Safety-ized for python.
+
+static
+void safe_set_item(G3TimesampleMap &self, std::string key, G3FrameObjectPtr value)
+{
+	int check_len = g3_vect_test_and_size(value);
+	if (check_len < 0) {
+		std::ostringstream s;
+		s << "Cannot add member (" << key << "): "
+		  << "not a supported vector type.";
+		throw g3timesample_exception(s.str());
+	}
+	if (check_len != self.times.size()) {
+		std::ostringstream s;
+		s << "Cannot add member (" << key << "): "
+		  << "not the same length as .times.";
+		throw g3timesample_exception(s.str());
+	}
+	self[key] = value;
+}
+
+static
+void safe_set_times(G3TimesampleMap &self, G3VectorTime _times)
+{
+	// Only allow this if it doesn't upset consistency.  We will
+	// assume that, coming in, we're internally consistent.
+	if (_times.size() != self.times.size() && self.size() != 0) {
+		std::ostringstream s;
+		s << "Cannot set .times because it conflicts with "
+		  << "the established number of samples (" << self.times.size() << ").";
+		throw g3timesample_exception(s.str());
+	}
+	self.times = _times;
+}
+
+static
+G3VectorTime safe_get_times(const G3TimesampleMap &self)
```

Comment:
Should return a reference `const G3VectorTime &` here, maybe?  I am also possibly being too obsessive about memory use.

Reply:
Solution found... turns out you can pass a reference to the data member in add_property so that it acts just like (the set part of) def_readwrite!